torchchat/cli/cli.py

```diff
@@ -498,9 +497,10 @@ def _add_speculative_execution_args(parser) -> None:
 
 
 def arg_init(args):
-    if not (torch.__version__ > "2.3"):
+    torch_version = importlib.metadata.version("torch")
+    if not torch_version or (torch_version <= "2.3"):
         raise RuntimeError(
-            f"You are using PyTorch {torch.__version__}. At this time, torchchat uses the latest PyTorch technology with high-performance kernels only available in PyTorch nightly until the PyTorch 2.4 release"
+            f"You are using PyTorch {torch_version}. At this time, torchchat uses the latest PyTorch technology with high-performance kernels only available in PyTorch nightly until the PyTorch 2.4 release"
```

Comment:
We might defer that to commands that actually use torch,  and then we can just use torch.__version__ ?  No point in raising a version issue when all we do is run help or similar non-model cli commands? 

Wdyt @Jack-Khuu

Reply:
I like having it here, since we can catch it at the very beginning as opposed to having version checking in every file. 

Doing it this way will speed up the interface by a lot. Right now --help and --command is super slow. It also bogs down CI.

Reply:
I could see it either way, but I think I agree with @byjlw. I don't think the non-running commands are valuable in-and-of-themselves, so I think the argument would be that this is really a property of the tool (there is no `torchchat` without `torch`) and doing the lazy imports is just a speed optimization.

If there were standalone commands that didn't need `torch` (i.e. if people start using `download` just to fetch models, but then run them with something else), then I could see making this a soft requirement and deferring it to the commands that need it.

Reply:
I actually added what @mikekgfb is suggesting a while back, so this check being in `arg_init` actually does what all 3 y'all suggest (check only if needed, and check in one place)

https://github.com/pytorch/torchchat/blob/4a7dab8cfb7111aa2323ad840cda68d65b81e86f/torchchat.py#L62-L63

`arg_init` is only called when it is pertinent